service/controller/resource/clustersecret/openstack.go

```diff
@@ -58,10 +58,16 @@ func (r *Resource) generateOpenStackCloudConfig(ctx context.Context, cluster cap
 		return nil, microerror.Mask(invalidConfigError)
 	}
 
-	networkID := cluster.Status.Network.ID
-	subnetID := cluster.Status.Network.Subnet.ID
-	floatingNetworkID := cluster.Status.ExternalNetwork.ID
-	publicNetworkName := cluster.Status.ExternalNetwork.Name
+	var networkID, subnetID, floatingNetworkID, publicNetworkName string
+
+	if cluster.Status.Network != nil {
```

Comment:
Nice! This should fix the nil panic we've seen from time to time.

Reply:
Thanks and I'd actually like to move these to key functions. I'll do a follow up for that.